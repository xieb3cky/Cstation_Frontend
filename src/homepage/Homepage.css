@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@400;600;700&display=swap");

#background-video {
    width: 100vw;
    height: 100vh;
    object-fit: cover;
    position: fixed;
    left: 0;
    right: 0;
    /* top: 10vh; */
    bottom: 0;
    z-index: -1;
}

/* Start Landing Page */

.landing-page {
    width: 100vw;
    height: 90vh;
    position: fixed;
    /* background: linear-gradient(rgb(126, 147, 167), rgba(126, 147, 167, 0.616)); */
}


.landing-container {
    width: 90%;
    height: 70%;
    /* padding-right: 15px; */
    padding-left: 3rem;
    margin: 5rem;
}


.landing-page .info {
    width: 35%;
    float: left;
    margin-top: 130px;
}

.landing-page .info h1 {
    height: 150px;
    width: 500px;
    font-size: 3rem;
    margin: 0 0 20px;
    line-height: 1.5;
    color: white;
}

.landing-page .info p {
    margin: 0;
    line-height: 1.5;
    font-size: 1.25rem;
    color: #d5cece;
}

.or-option {
    font-size: 1.5rem;
    padding-top: 15px;
    padding-left: 50px;
    font-weight: 600;
    color: #f6e198;
}


.landing-page .info button {
    border: 0;
    font-family: inherit;
    font-size: inherit;
    font-weight: 600;
    border-radius: 20px;
    margin-top: 15px;
    padding: 10px 18px;
    cursor: pointer;
    color: #FFF;
    background-color: #6c63ff;
}

.landing-page .info button:hover {
    background: #f6e198;
    color: black;
    text-decoration: none;
}

.landing-page .image {
    width: 50%;
    float: right;
    margin-top: 35px;
}

.landing-page .image img {
    max-width: 100%;
}

.clearfix {
    clear: both;
}



/* End Landing Page */