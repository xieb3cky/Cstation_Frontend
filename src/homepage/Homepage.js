import React, { useContext, useState, useEffect } from "react";
// import { Link, NavLink } from "react-router-dom";
import "./Homepage.css";
import Map from "../search/Map";
import UserContext from "../auth/UserContext";
import video from "./ev-video2.mp4";
import CstationAPI from "../api/api";
import StationCard from "../stations/StationCard";
import "../search/Map.css"

/** Homepage of site.
 *
 * Shows welcome message or login/register buttons.
 *
 * Routed at /
 *
 * Routes -> Homepage
 */

function Homepage({ search }) {
    const { currUser, favStationInfo } = useContext(UserContext);

    // const [favStationInfo, setFavStations] = useState([]);

    console.debug("Homepage", "currentUser=", currUser);
    // Get charging station info for each of user's favorites
    // useEffect(() => {
    //     async function getFavStations() {
    //         if (currUser) {
    //             const res = await CstationAPI.getAllFavorites(currUser.id)
    //             setFavStations(res)
    //         }
    //     }
    //     getFavStations()
    // }, [currUser])
    // console.log(favStationInfo.result)
    // favStationInfo.result.map((s) => {
    //     console.log(s["id"])
    //     console.log(s["name"])
    // })


    // useEffect(() => {

    //     getFavStationInfo();
    //     console.log(favStationInfo)


    // }, [])



    // console.log(favorites)
    const loggedInHome = () => {
        return <div>
            {/* <div className="fullmap">
                <Map />
            </div> */}

            <div className="citems">

                {favStationInfo.result &&
                    <div className="">
                        {favStationInfo.result.map(s => (
                            <StationCard
                                key={s.id}
                                id={s.id}
                                name={s.name}
                                address={s.address}
                                phone={s.phone}
                                email={s.email}
                                available={s.available}
                                charger_type={s.charger_type}
                                lat={s.lat}
                                long={s.long}
                            />
                        ))}
                    </div>
                }
            </div>

        </div>
    }



    const loggedOutHome = () => {
        return <div>
            <video id="background-video" autoPlay loop muted>
                <source src={video} type="video/mp4" />
            </video>
            <div class="landing-page">
                <div class="landing-container">

                    <div class="info">
                        <h1>Search Electric Chargers Near You</h1>
                        <p>1. Enter your address</p>
                        <p>2. Select charger type</p>
                        <p>3. Select max result</p>
                        <p>4. Hit Search</p>
                        <div className="or-option"> OR </div>
                        <button>Quick Search</button>
                    </div>
                </div>
            </div>
        </div>
    }

    return (
        <>
            {currUser && favStationInfo ? loggedInHome() : loggedOutHome()}
        </>
    );
}

export default Homepage;

