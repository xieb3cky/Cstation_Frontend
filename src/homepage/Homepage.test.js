import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Homepage from "./Homepage";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Homepage />
      </UserProvider>
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
