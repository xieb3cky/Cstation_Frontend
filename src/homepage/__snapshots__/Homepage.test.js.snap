// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <video
      autoplay=""
      id="background-video"
      loop=""
    >
      <source
        src="ev-video2.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="landing-page"
    >
      <div
        class="landing-container"
      >
        <div
          class="info"
        >
          <h1>
            Search Electric Car Chargers Near You
          </h1>
          <p>
            1. Enter your address
          </p>
          <p>
            2. Select charger type
          </p>
          <p>
            3. Select max result
          </p>
          <p>
            4. Hit Search
          </p>
          <div
            class="or-option"
          >
             OR 
          </div>
          <button>
            Quick Search
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    <video
      autoplay=""
      id="background-video"
      loop=""
    >
      <source
        src="ev-video2.mp4"
        type="video/mp4"
      />
    </video>
    <div
      class="landing-page"
    >
      <div
        class="landing-container"
      >
        <div
          class="info"
        >
          <h1>
            Search Electric Car Chargers Near You
          </h1>
          <p>
            1. Enter your address
          </p>
          <p>
            2. Select charger type
          </p>
          <p>
            3. Select max result
          </p>
          <p>
            4. Hit Search
          </p>
          <div
            class="or-option"
          >
             OR 
          </div>
          <button>
            Quick Search
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
